tests/client/test_utils.py

```diff
@@ -0,0 +1,94 @@
+import os
+import pytest
+from codecs import encode
+
+import mock
+
+from scrapinghub.utils import parse_auth
+from scrapinghub.utils import format_iter_filters
+
+
+def test_format_iter_filters():
+    # work with empty params
+    assert format_iter_filters({}) == {}
+
+    # doesn't affect other params
+    params = {'a': 123, 'b': 456}
+    assert format_iter_filters(params) == params
+
+    # pass filter as-is if not list
+    params = {'filter': 'some-string'}
+    assert format_iter_filters(params) == params
+
+    # work fine with empty filter
+    params = {'filter': []}
+    assert format_iter_filters(params) == params
+
+    # pass string filters as-is
+    params = {'filter': ['str1', 'str2']}
+    assert format_iter_filters(params) == params
+
+    # converts list-formatted filters
+    params = {'filter': [['field', '>=', ['val']], 'filter2']}
+    assert (format_iter_filters(params) ==
+            {'filter': ['["field", ">=", ["val"]]', 'filter2']})
+
+    # works the same with tuple entries
+    params = {'filter': [('field', '==', ['val'])]}
+    assert (format_iter_filters(params) ==
+            {'filter': ['["field", "==", ["val"]]']})
+
+    # exception if entry is not list/tuple or string
+    with pytest.raises(ValueError):
+        format_iter_filters({'filter': ['test', 123]})
+
+
+def test_parse_auth_none():
+    with pytest.raises(RuntimeError):
+        parse_auth(None)
+
+
+@mock.patch.dict(os.environ, {'SH_APIKEY': 'testkey'})
+def test_parse_auth_none_with_env():
+    assert parse_auth(None) == ('testkey', '')
+
+
+def test_parse_auth_tuple():
+    assert parse_auth(('test', 'test')) == ('test', 'test')
+    assert parse_auth(('apikey', None)) == ('apikey', None)
+
+    with pytest.raises(ValueError):
+        parse_auth(('user', 'pass', 'bad-param'))
+
+    with pytest.raises(ValueError):
+        parse_auth((None, None))
+
+    with pytest.raises(ValueError):
+        parse_auth((1234, ''))
+
+
+def test_parse_auth_not_string():
+    with pytest.raises(ValueError):
+        parse_auth(12345)
+
+
+def test_parse_auth_simple():
+    assert parse_auth('user:pass') == ('user', 'pass')
+
+
+def test_parse_auth_bad_apikey():
+    with pytest.raises(ValueError):
+        parse_auth('apikey')
+
+
+def test_parse_auth_apikey():
+    test_key = u'\xe3\x98\xb8\xe6\x91\x84\xe9'
+    apikey = encode(test_key.encode('utf8'), 'hex_codec').decode('ascii')
```

Comment:
I tried to show that apikey is just a utf8 string encoded with hex: all the conversions are related with that `codecs.encode` accepts a binary and returns a binary. But yeah, now I see it's not clear, let's just test with pre-generated test apikey